File: Leetcode 2001-2100/2029/2029.java
// Leetcode 2029: Stone Game IX
// https://leetcode.com/problems/stone-game-ix/
// Solved on 5th of March, 2026
class Solution {
    /**
     * Determines if Alice can win the stone game given the rules of modulo 3 sums.
     * 
     * @param stones An array of integers representing the values of the stones.
     * @return true if Alice has a winning strategy, false otherwise.
     */
    public boolean stoneGameIX(int[] stones) {
        int[] counts = new int[3];
        for (int stine : stones) {
            counts[stone % 3]++;
        }
        
        if (counts[0] % 2 == 0) {
            return counts[1] > 0 && counts[2] > 0;
        } else {
            return Math.abs(counts[1] - counts[2]) > 2;
        }
    }
}

File: Leetcode 2001-2100/2029/2029.py
# Leetcode 2029: Stone Game IX
# https://leetcode.com/problems/stone-game-ix/
# Solved on 5th of March, 2026
class Solution:
    def stoneGameIX(self, stones: list[int]) -> bool:
        """
        Determines if Alice can win the stone game given the rules of divisibility by 3.

        :param stones: A list of integers representing the values of the stones.
        :return: True if Alice wins, False otherwise.
        """
        remainderCounts = [0, 0, 0]
        for stoneValue in stones:
            remainderCounts[stoneValue % 3] += 1

        if remainderCounts[0] % 2 == 0:
            return remainderCounts[1] > 0 and remainderCounts[2] > 0

        return abs(remainderCounts[1] - remainderCounts[2]) > 2